quinn-proto/src/endpoint.rs

```diff
@@ -104,6 +104,9 @@ impl Endpoint {
         loop {
             let &ch = self.dirty_conns.iter().next()?;
             loop {
+                while let Some(event) = self.connections[ch].poll_endpoint_events() {
+                    self.handle_event(ch, event);
+                }
```

Comment:
The tokio bindings currently assume that application-facing events are only queued by `timeout` and `handle`; outgoing IO is not expected to expose new information to the application. I see that putting this in `poll` would be awkward wrt. the test cod as it stands; as a temporary measure until the decoupling is complete, maybe we could instead put it in a method that gets called by both `handle` and `timeout`?